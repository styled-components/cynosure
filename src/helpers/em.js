// @flow
/** @module Helpers */

import stripUnit from './stripUnit'

/**
 * Convert pixel value to ems. The default base value is 16px, but can be changed with the second argument
 * @static
 * @param {string|number} pxval - The pixel value you want to convert
 * @param {string|number} [base = "16px"] - The base size to convert from
 * @return {String} The converted value
 * @example
 * // Styles as object usage
 * const styles = {
 *   'height': em('16px')
 * }
 *
 * // styled-components usage
 * const div = styled.div`
 *   height: ${em('16px')}
 * `
 *
 * // CSS in JS Output
 *
 * element {
 *   'height': '1em'
 * }
 */

/* eslint-disable no-param-reassign */
function em(pxval: string|number, base?: string|number = '16px') {
  if (typeof pxval === 'string') {
    pxval = stripUnit(pxval)
  }
  if (base && typeof base === 'string') {
    base = stripUnit(base)
  }

  // $FlowFixMe
  return `${pxval / base}em`
}

export default em
