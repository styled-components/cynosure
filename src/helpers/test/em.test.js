// @flow
import em from '../em'

describe('em', () => {
  it('should convert a simple number to ems', () => {
    expect({ height: em(16) }).toEqual({
      height: '1em',
    })
  })
})
