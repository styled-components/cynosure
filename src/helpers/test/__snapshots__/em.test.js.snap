// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`em should convert a simple number to ems 1`] = `
Object {
  "height": "1em",
}
`;
